pkg/collect/copy_from_host.go

```diff
@@ -229,6 +230,13 @@ func copyFromHostCreateDaemonSet(ctx context.Context, client kubernetes.Interfac
 	for {
 		select {
 		case <-time.After(1 * time.Second):
+			if !retryFailedMount {
```

Comment:
Shouldn't this logic be in line `458`? When we check the status of a pod, we want to exit with error when event is `FailedMount` and `retryFailedMount == false`. The implementation you have here will never check the pod status when `retryFailedMount == false`.

```go
for _, event := range events.Items {
	// If the pod has a FailedMount event, it means that the pod failed to mount the volume and the pod will be stuck in the Pending state.
	// In this case, we return an error to the caller to indicate that path does not exist.
	if event.Reason == "FailedMount" && !retryFailedMount {
		klog.V(2).Infof("pod %s has a FailedMount event: %s", pod.Name, event.Message)
		return errors.Errorf("%q path does not exist", hostPath)
	}
}
```

Reply:
good catch, I have moved that into function and updated the tests